jupyterlab_server/listings_handler.py

```diff
@@ -0,0 +1,73 @@
+"""Tornado handlers for dynamic listings loading."""
+
+# Copyright (c) Jupyter Development Team.
+# Distributed under the terms of the Modified BSD License.
+
+import os
+import time
+import json
+import re
+
+from .server import APIHandler
+
+import requests
+
+
+LISTINGS_URL_SUFFIX='@jupyterlab/extensionmanager-extension/listings.json'
+
+
+def fetch_listings():
+    if len(ListingsHandler.blacklist_uris) > 0:
+        blacklist = []
+        for blacklist_uri in ListingsHandler.blacklist_uris:
+            print('Fetching blacklist from {}'.format(ListingsHandler.blacklist_uris))
```

Comment:
I guess switch this to use logging?

Reply:
This method is accessed from 2 places. I have passed the logger from one place, but have issue to pass it from the other place (in `handlers.py`)

I have added a TODO and will double check for this. At worth, it will use print the first time it fetches the listings, and logger the next times.

Reply:
Currently this logging stands out a bit since it uses a print statement:

<img width="1024" alt="Screen Shot 2020-03-18 at 7 48 31 AM" src="https://user-images.githubusercontent.com/1186124/76957814-de567500-68ec-11ea-8def-32ca2e8ea22c.png">

Hm... Looking at it briefly, I believe you could make this an instance method of the `ListingsHandler`, so that you can use `self.log` in the body of this function.

Then in the `ioloop.PeriodicCallback`, just pass in `self.fetch_listings` instead. Also maybe you could just call `self.fetch_listings()` in `initialize` instead of in `handlers.py`?

Reply:
The latest push will only use print on start, then the periodic pull will use the log.

For now I am initializing via a class method to ensure the list are loaded on server start. At this point, the self.log is not available. After thee PeriodicCallback passes the self.log and the console will look good.

One option would be to not call the fetch_listing on server start, but this would penalize the first user as this first user would need for the HTTP request of the server to the listings servers.

I guess I could copy the code sections that create the logger when self.log is not available. Let me try that.

Reply:
All good. Now always using logger.

<img width="559" alt="Screenshot 2020-03-18 at 13 45 29" src="https://user-images.githubusercontent.com/226720/76962205-18d90580-691f-11ea-9ca5-1f8752b4d4ac.png">

Reply:
Great!